model/common/src/icon4py/model/common/metrics/metric_fields.py

```diff
@@ -881,205 +774,104 @@ def compute_flat_idx(
 ) -> fa.EdgeField[wpfloat]:
     extrapol_dist = 5.0
     z_aux1 = maximum(z_ifc(E2C[0]), z_ifc(E2C[1]))
     z_aux2 = z_aux1 - extrapol_dist
-
     return z_aux2
 
 
-@program(grid_type=GridType.UNSTRUCTURED)
-def compute_z_aux2(
-    z_ifc_sliced: fa.CellField[wpfloat],
-    z_aux2: fa.EdgeField[wpfloat],
-    horizontal_start: int32,
-    horizontal_end: int32,
-):
-    _compute_z_aux2(
-        z_ifc=z_ifc_sliced, out=z_aux2, domain={dims.EdgeDim: (horizontal_start, horizontal_end)}
-    )
-
-
 @field_operator
-def _compute_pg_edgeidx_vertidx(
-    c_lin_e: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2CDim], float],
-    z_ifc: fa.CellKField[wpfloat],
+def _compute_pressure_gradient_downward_extrapolation_mask_distance(
+    z_mc: fa.CellKField[wpfloat],
+    c_lin_e: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2CDim], wpfloat],
     z_ifc_sliced: fa.CellField[wpfloat],
     e_owner_mask: fa.EdgeField[bool],
-    flat_idx_max: fa.EdgeField[int32],
-    e_lev: fa.EdgeField[int32],
-    k_lev: fa.KField[int32],
-    pg_edgeidx: fa.EdgeKField[int32],
-    pg_vertidx: fa.EdgeKField[int32],
-) -> tuple[fa.EdgeKField[int32], fa.EdgeKField[int32]]:
-    z_aux2 = _compute_z_aux2(z_ifc_sliced)
+    flat_idx_max: fa.EdgeField[gtx.int32],
+    e_lev: fa.EdgeField[gtx.int32],
+    k_lev: fa.KField[gtx.int32],
+    horizontal_start_distance: int32,
+    horizontal_end_distance: int32,
+) -> tuple[fa.EdgeKField[bool], fa.EdgeKField[wpfloat]]:
+    """
+    Compute an edge mask and extrapolation distance for grid points requiring downward extrapolation of the pressure gradient.
+
+    See pg_edgeidx and pg_exdist in mo_vertical_grid.f90
+
+    Args:
+        z_mc:
+        c_lin_e:
+        z_ifc_sliced:
+        e_owner_mask:
+        flat_idx_max:
+        e_lev:
+        k_lev:
+        horizontal_start_distance:
+        horizontal_end_distance:
+
+    Returns:
+        edge index mask for points requiring downward extraplation
+        extrapolation distance
+
+    """
+
     e_lev = broadcast(e_lev, (dims.EdgeDim, dims.KDim))
     k_lev = broadcast(k_lev, (dims.EdgeDim, dims.KDim))
-    z_mc = average_cell_kdim_level_up(z_ifc)
     z_me = _cell_2_edge_interpolation(in_field=z_mc, coeff=c_lin_e)
+    downward_distance = _compute_downward_extrapolation_distance(z_ifc_sliced)
+    z_aux3 = where(
+        (e_lev >= horizontal_start_distance) & (e_lev < horizontal_end_distance),
+        downward_distance,
+        0.0,
+    )
     pg_edgeidx = where(
-        (k_lev >= (flat_idx_max + 1)) & (z_me < z_aux2) & e_owner_mask, e_lev, pg_edgeidx
+        (k_lev >= (flat_idx_max + 1)) & (z_me < downward_distance) & e_owner_mask, e_lev, 0
     )
     pg_vertidx = where(
-        (k_lev >= (flat_idx_max + 1)) & (z_me < z_aux2) & e_owner_mask, k_lev, pg_vertidx
+        (k_lev >= (flat_idx_max + 1)) & (z_me < downward_distance) & e_owner_mask, k_lev, 0
     )
-    return pg_edgeidx, pg_vertidx
-
+    pg_edge_mask = where((pg_edgeidx > 0) & (pg_vertidx > 0), True, False)
```

Comment:
could we do something like this?
```suggestion
    pg_edge_mask = (pg_edgeidx > 0) and (pg_vertidx > 0)
```